test/test_fcntl.rs

```diff
@@ -232,3 +232,46 @@ mod test_posix_fadvise {
         assert_eq!(errno, Errno::ESPIPE as i32);
     }
 }
+
+#[cfg(any(target_os = "linux",
+          target_os = "android",
+          target_os = "emscripten",
+          target_os = "fuchsia",
+          any(target_os = "wasi", target_env = "wasi"),
+          target_env = "freebsd"))]
+mod test_posix_fallocate {
+
+    use tempfile::NamedTempFile;
+    use std::{io::Read, os::unix::io::{RawFd, AsRawFd}};
+    use nix::errno::Errno;
+    use nix::fcntl::*;
+    use nix::unistd::pipe;
+
+    #[test]
+    fn test_success() {
+        const LEN: usize = 100;
+        let mut tmp = NamedTempFile::new().unwrap();
+        let fd = tmp.as_raw_fd();
+        let res = posix_fallocate(fd, 0, LEN as libc::off_t).unwrap();
+        let mut data = [1u8; LEN];
+        assert_eq!(tmp.read(&mut data).expect("read failure"), LEN);
+        assert_eq!(&data as &[u8], &[0u8; LEN] as &[u8]);
+        assert_eq!(res, 0);
+    }
+
+    #[test]
+    fn test_errno() {
+        let (rd, _wr) = pipe().unwrap();
+        let errno = posix_fallocate(rd as RawFd, 0, 100).unwrap();
+        match Errno::from_i32(errno) {
+            Errno::EBADF
```

Comment:
Can't you narrow this list down?  I don't think there are very many different error codes that could be returned in this case.

Reply:
Sure!